### PR TITLE
BC-8404 - Move tldraw config to issolate it from server config

### PR DESCRIPTION
**Pull request by SevenWaysDP**
# Description

## Links to Tickets or other pull requests

- https://ticketsystem.dbildungscloud.de/browse/BC-8404

- https://github.com/hpi-schul-cloud/tldraw-client/pull/118
- https://github.com/hpi-schul-cloud/dof_app_deploy/pull/1039

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: ansible/roles/tldraw-server/templates/configmap.yml.j2 [hunk 1]
```diff
@@ -6,16 +6,13 @@ metadata:
   labels:
     app: tldraw-server
 data:
-  API_HOST: "http://api-svc:3030"
-  WS_PATH_PREFIX: "/tldraw-server"
-  WS_PORT: "3345"
```
Comment (mamutmk5) [inline ansible/roles/tldraw-server/templates/configmap.yml.j2:11]: What happend to the WS_PORT?

File: ansible/roles/tldraw-server/templates/configmap.yml.j2 [hunk 2]
```diff
@@ -6,16 +6,13 @@ metadata:
   labels:
     app: tldraw-server
 data:
-  API_HOST: "http://api-svc:3030"
```
Comment (mamutmk5) [inline ansible/roles/tldraw-server/templates/configmap.yml.j2:9]: What happend to the API_HOST?

File: ansible/roles/tldraw-server/templates/configmap.yml.j2 [hunk 3]
```diff
@@ -6,16 +6,13 @@ metadata:
...
+  METRICS_COLLECT_DEFAULT: "{{ TLDRAW_METRICS_COLLECT_DEFAULT }}"
   REDIS_CLUSTER_ENABLED: "true"
   REDIS_SENTINEL_SERVICE_NAME: "valkey-headless.{{ NAMESPACE }}.svc.cluster.local"
-  TLDRAW__WEBSOCKET_URL: "wss://{{ DOMAIN }}/tldraw-server"
-  TLDRAW__ASSETS_ENABLED: "true"
-  TLDRAW__ASSETS_MAX_SIZE_BYTES: "10485760"
-  TLDRAW__ASSETS_ALLOWED_MIME_TYPES_LIST: "image/png,image/jpeg,image/gif,image/svg+xml"
-  FEATURE_TLDRAW_ENABLED: "true"
-  NEST_LOG_LEVEL: "info"
-  EXIT_ON_ERROR: "true"
```
Comment (mamutmk5) [inline ansible/roles/tldraw-server/templates/configmap.yml.j2:21]: no more exit on error?

Comment (SevenWaysDP) [reply ansible/roles/tldraw-server/templates/configmap.yml.j2:21]: Is default true in code

File: ansible/roles/tldraw-server/templates/configmap.yml.j2 (hunk 1 above)
Comment (SevenWaysDP) [reply ansible/roles/tldraw-server/templates/configmap.yml.j2:11]: Is default in code

File: ansible/roles/tldraw-server/templates/configmap.yml.j2 (hunk 2 above)
Comment (SevenWaysDP) [reply ansible/roles/tldraw-server/templates/configmap.yml.j2:9]: rename to AUTHORIZATION_API_HOST

File: src/modules/server/server.config.ts [hunk 4]
```diff
@@ -3,28 +3,28 @@ import { IsArray, IsBoolean, IsNumber, IsString, IsUrl } from 'class-validator';
 
 export class ServerConfig {
```
Comment (bergatco) [inline src/modules/server/server.config.ts:4]: Können wir `ServerConfig` umbenennen? Als ich das im Code gelesen habe, haben bei mir gleich alle Alarmglocken geläutet, weil ich dachte, wir nutzen hier die `ServerConfig` vom `schulcloud-server`.
Mein Vorschlag wäre `TldrawServerConfig`?
Wenn nicht, ist auch okay! Muss ich nur meinen inneren Monk versuchen das beizubringen.

**Review by bergatco — APPROVED** [review]

File: ansible/roles/tldraw-server/templates/configmap.yml.j2 (hunk 3 above)
Comment (mamutmk5) [reply ansible/roles/tldraw-server/templates/configmap.yml.j2:21]: After all i personally think it's a very bad Idea to exit on error.
Specaly on a multi user and board POD.
Why i think so is that if for one user an error happened all other users with other boards are even face the consequences of the restart.

**Review by mamutmk5 — APPROVED** [review]

**Pull request merged**